### PR TITLE
hoist precomputed exprs from indices

### PR DESCRIPTION
**Pull request by ngimel**
This generates compilable code for maskrcnn graph 13, with ceilings hoisted to be computed on the host. But it now fails with 
```
  File "/scratch/ngimel/work/pytorch/torch/_dynamo/symbolic_convert.py", line 379, in wrapper
    self.output.compile_subgraph(self, reason=reason)
  File "/scratch/ngimel/work/pytorch/torch/_dynamo/output_graph.py", line 562, in compile_subgraph
    pass1.foreach(stack_values)
  File "/scratch/ngimel/work/pytorch/torch/_dynamo/codegen.py", line 166, in foreach
    self(i)
  File "/scratch/ngimel/work/pytorch/torch/_dynamo/codegen.py", line 148, in __call__
    output.extend(value.reconstruct(self))
  File "/scratch/ngimel/work/pytorch/torch/_dynamo/variables/dicts.py", line 40, in reconstruct
    codegen.create_load_python_module(collections),
TypeError: create_load_python_module() missing 1 required positional argument: 'push_null'

from user code:
   File "/scratch/ngimel/work/env/lib/python3.9/site-packages/torchvision-0.15.0a0+928b05c-py3.9-linux-x86_64.egg/torchvision/models/detection/backbone_utils.py", line 58, in forward
    x = self.fpn(x)
```
looks like we never execute this `create_load_python_module()` path for other subgraphs. 
Any advice on how to fix this @voznesenskym @jansel ?

cc @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

File: torch/_inductor/codegen/triton.py [hunk 1]
```diff
@@ -860,7 +864,7 @@ def indexing(
                 # indirect indexing
                 cse_var = self.cse.varname_map[var.name]
                 mask_vars.update(cse_var.mask_vars)
-            elif var.name.startswith("s"):
+            elif var.name.startswith("s") or var.name.startswith("ps"):
```
Comment (Skylion007) [inline torch/_inductor/codegen/triton.py:867]:
```suggestion
            elif var.name.startswith(("s", "ps")):
```
More readable and efficient.

**Comment by jansel** [comment]
I think the 'push_null' error is related to @williamwen42's Python 3.11 work.

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**